sync/src/synchronizer/mod.rs

```diff
@@ -257,25 +257,26 @@ impl<CS: ChainStore> Synchronizer<CS> {
                     .remove_blocks_by_parent(&block.header().hash());
                 for block in pre_orphan_block {
                     let block = Arc::new(block);
-                    if self
+                    let ret = if self
```

Comment:
we can use rust idiomatic for option and result, it's easier to read:
```
if let Some(Err(e)) = self.shared.block_header(&block.header().parent_hash())
    .and(|| self.accept_block(peer, &block).err())
    .or(|| Err(err_msg(...))) {
     debug!(...)
}
```

Reply:
I have tried to change it more rust idiomatic but failed...
Could you give me the complete code snips? Thanks ~

Reply:
Could you try this?  my previous comment is something wrong, sorry.
```
                    match self
                        .shared
                        .block_header(&block.header().parent_hash())
                        .and(Some(self.accept_block(peer, &block)))
                    {
                        None => {
                            debug!(
                                target: "sync", "[Synchronizer] accept orphan block {:?}, parent not found",
                                block
                            );
                        }
                        Some(Err(e)) => {
                            debug!(
                                target: "sync", "[Synchronizer] accept orphan block {:?} error {:?}",
                                block,
                                e
                            );
                        }
                        Some(Ok(_)) => {}
                    }
```

Reply:
Thanks.
I have totally rewritten this function. PTAL